src/Http/Routing/src/RouteEndpointBuilder.cs

```diff
@@ -79,9 +80,46 @@ public override Endpoint Build()
             requestDelegate,
             RoutePattern,
             Order,
-            new EndpointMetadataCollection(Metadata),
+            CreateMetadataCollection(Metadata),
             DisplayName);
 
         return routeEndpoint;
     }
+
+    private static EndpointMetadataCollection CreateMetadataCollection(IList<object> metadata)
+    {
+        if (metadata is { Count: > 0 })
+        {
+            var hasCorsMetadata = false;
+            HttpMethodMetadata? httpMethodMetadata = null;
+
+            // Before create the final collection we
+            // need to update the IHttpMethodMetadata if
+            // a CORS metadata is present
+            for (var i = 0; i < metadata.Count; i++)
+            {
+                if (metadata[i] is HttpMethodMetadata methodMetadata)
+                {
+                    // Storing only the last entry
+                    // since the last metadata is the most significant.
+                    httpMethodMetadata = methodMetadata;
+                }
+                else if (!hasCorsMetadata && metadata[i] is ICorsMetadata)
+                {
+                    // IEnableCorsAttribute, IDisableCorsAttribute and ICorsPolicyMetadata
+                    // are ICorsMetadata
+                    hasCorsMetadata = true;
```

Comment:
Should DisableCors set `AcceptCorsPreflight` to true? Does the cors middleware have behavior we'd prefer over the fake 405 endpoint in that case?

Reply:
> Should DisableCors set AcceptCorsPreflight to true?

I was concerned about that, however that is basically how MVC detects the CORS metadata.

> Does the cors middleware have behavior we'd prefer over the fake 405 endpoint in that case?

Not sure if preferable or not (MVC endpoints will no produce 405 in this case), but yes the middleware handle the `IDisableCorsAttribute` and complete the `OPTIONS` request with `204 NO Content` without the expected headers

Reply:
Also, since we don't have `CORS` attributes in the routing layer, we could not determine if it is `Disable` or `Enable` at this layer :(.

Reply:
Also, in my opinion make more sense the routing delegates the decision to fail the `OPTIONS` request to the CORS middleware, since CORS is present. And only fail as 405 when it is truly not able to handle (Not using CORS). Is that make sense?